scripts/build.mjs

```diff
@@ -0,0 +1,228 @@
+import { resolve } from 'path';
+import { readFileSync, writeFileSync } from 'fs';
+import { execSync } from 'child_process';
+import { build } from 'vite';
+
+const pwd = process.cwd();
+
+const options = {
+  srcDir: resolve(pwd, 'src'),
+  outDir: resolve(pwd, 'dist'),
+  assets: ['_locales', 'assets', 'vendor/@whotracksme/ui/src/images'],
+};
+
+// Generate arguments from command line
+const argv = process.argv.slice(2).reduce((acc, arg, index, arr) => {
+  if (arg.startsWith('--')) {
+    const key = arg.slice(2);
+    const value = arr[index + 1];
+    acc[key] = !value || value.startsWith('--') ? true : value;
+  }
+  return acc;
+}, {});
+
+const manifest = JSON.parse(
+  readFileSync(
+    resolve(options.srcDir, `manifest.${argv.target || 'chromium'}.json`),
+    'utf8',
+  ),
+);
+
+const config = {
+  configFile: false,
+  root: process.cwd(),
+  plugins: [
+    // Required for disabling module preload feature
+    {
+      name: 'remove-vite-build-import-analysis',
+      enforce: 'pre',
+      options: (options) => {
+        options.plugins = options.plugins.filter(
+          (p) => p.name !== 'vite:build-import-analysis',
+        );
+        return options;
+      },
+    },
+    // Required for cleaning up scripts tag generation in html files
+    {
+      name: 'clean-imports-for-html-generation',
+      generateBundle(options, bundle) {
+        Object.values(bundle).forEach((chunk) => {
+          if (
+            chunk.type === 'chunk' &&
+            !chunk.isEntry &&
+            !chunk.facadeModuleId?.endsWith('.html') &&
+            chunk.imports &&
+            chunk.imports.length
+          ) {
+            chunk.imports = [];
+          }
+        });
+      },
+    },
+    // Required for correct html assets path because of the root dir
+    {
+      name: 'clean-html-src-path',
+      enforce: 'post',
+      generateBundle: (options, bundle) => {
+        Object.values(bundle).forEach((chunk) => {
+          if (
+            chunk.type === 'asset' &&
+            chunk.fileName.endsWith('.html') &&
+            !chunk.fileName.includes('node_modules')
+          ) {
+            chunk.fileName = chunk.fileName.replace('src/', '');
+          }
+        });
+      },
+    },
+  ],
+  resolve: {
+    alias: [
+      { find: '/hybrids.js', replacement: 'hybrids' },
+      { find: '/vendor', replacement: '/src/vendor' },
+    ],
+  },
+  build: {
+    outDir: options.outDir,
+    emptyOutDir: false,
+    minify: false,
+    polyfillModulePreload: false,
+    watch: argv.watch ? {} : null,
+  },
+};
+
+// --- Generate dist structure ---
+
+// generate dist folder
+execSync(`rm -rf ${options.outDir} && mkdir -p ${options.outDir}`);
+
+// copy static assets
+options.assets.forEach((path) => {
+  execSync(`mkdir -p ${options.outDir}/${path}`);
```

Comment:
`rimraf` is used in another context, which build script does not change - copy of vendors and cosmetics in `postinstall` script from `manifest.json`. I am not a fan of adding dependencies where it is not required. 

The current vendor copy definitions use backslashes, and from quick look it does not support windows ;)

Reply:
Switching to some NodeJs API instead of running shell commands would be a good idea IMHO if possible.

Or at least, we should quote everything. For instance,
`mkdir -p ${options.outDir}/${path}` ->
`mkdir -p "${options.outDir}/${path}"` and
`rm -rf ${options.outDir}` ->
`rm -rf "${options.outDir}"`

Reply:
OK, I will add `shelljs` then.